### PR TITLE
chore: update go mod to remove requires

### PR DESCRIPTION
**Pull request by rsmitty**
This PR updates the go mod to use the latest v0.3 capi release, as well
as downgrade controller-runtime to match. The only real fallout of this
was the removal of the ctx from some calls to the k8s api, so shouldn't
be a big deal to readd this stuff once we move to v0.4 of capi.

Signed-off-by: Spencer Smith <spencer.smith@talos-systems.com>

### CODE REVIEW

File: pkg/capi/infrastructure/infrastructure.go [hunk 1]
```diff
@@ -24,7 +24,8 @@ type Provider interface {
 	WatchingNamespace() string
 	Configure(interface{}) error
 	PreInstall() error
-	IsInstalled(ctx context.Context, clientset *kubernetes.Clientset) (bool, error)
+	// TODO: readd ctx as input here once we bump up controller-runtime version.
+	IsInstalled(clientset *kubernetes.Clientset) (bool, error)
```
Comment (Unix4ever) [inline pkg/capi/infrastructure/infrastructure.go:28]: I would suggest to keep using the context in all the places and just skip passing it to underlying kubernetes call for now.

Comment (rsmitty) [reply pkg/capi/infrastructure/infrastructure.go:28]: It barfed during linting unfortunately :/

Comment (Unix4ever) [reply pkg/capi/infrastructure/infrastructure.go:28]: weird. Would be nice to keep it at least in the interface, but I guess it's fine either way.

Comment (rsmitty) [reply pkg/capi/infrastructure/infrastructure.go:28]: I can probably tell the linter to ignore. I'll poke at it again once I'm back home.

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**